crates/turborepo-lib/src/commands/login.rs

```diff
@@ -0,0 +1,106 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::{anyhow, Result};
+use axum::{extract::Query, response::Redirect, routing::get, Router};
+use log::{debug, warn};
+use serde::Deserialize;
+use tokio::sync::OnceCell;
+
+use crate::{
+    client::{APIClient, UserClient},
+    commands::CommandBase,
+    config::{default_user_config_path, UserConfigLoader},
+    get_version,
+    ui::{BOLD, CYAN},
+};
+
+const DEFAULT_HOST_NAME: &str = "127.0.0.1";
+const DEFAULT_PORT: u16 = 9789;
+
+pub async fn login(base: CommandBase) -> Result<()> {
+    let repo_config = base.repo_config()?;
```

Comment:
We need to add `--__test-run` otherwise the integration tests open up a browser and will hang for user input.